api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -244,12 +252,76 @@ private FirecloudWorkspace attemptFirecloudWorkspaceCreation(FirecloudWorkspaceI
 
     dbWorkspace.setBillingMigrationStatusEnum(BillingMigrationStatus.NEW);
 
-    dbWorkspace = workspaceService.getDao().save(dbWorkspace);
+    updateWorkspaceBillingAccount(dbWorkspace, workspace.getBillingAccountName());
+
+    try {
+      dbWorkspace = workspaceService.getDao().save(dbWorkspace);
+    } catch (Exception e) {
+      // Tell Google to set the billing account back to the free tier if the workspace creation
```

Comment:
If I am saving the workspace with the updates billing account, I believe I still need to make a try/catch call to update the db if the Google call fails.

Proposed
1. save db w/ new billing account (succeeds)
2. call google (fails)
3. save db w/ old billing account to keep state in sync OR delete the workspace b/c the create call didn't fully complete successfully.

Current
1. call google w/ new billing account (succeeds)
2. save db (fails)
3. call google w/ old billing account to keep state in sync

My reasoning for going with the current workflow was that I felt like a db save call is less likely to fail than an API call. There's also the effect of having a partially successful workspace entry in our database. I could see a case for not failing the entire call if the updateBillingInfo call fails since the entire call is quite expensive but I view that more of a feature add on that we can talk to product about.